tests/test_autodetect.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from staging_service.autodetect.GenerateMappings import type_to_extension_mapping
+from staging_service.AutoDetectUtils import AutoDetectUtils
+from staging_service.app import inject_config_dependencies
+from tests.test_utils import bootstrap_config
+
+
+@pytest.fixture(autouse=True)
+def run_before_and_after_tests(tmpdir):
+    config = bootstrap_config()
+    inject_config_dependencies(config)
+
+
+def test_config():
+    """
+    Test to make sure config injection is working for AutoDetectUtils
+    #TODO Can test PATH injections as well
+    :return:
+    """
+    mappings_fp = AutoDetectUtils._FILE_EXTENSION_MAPPINGS
+    mappings = AutoDetectUtils._MAPPINGS
+    assert mappings
+    assert mappings_fp
+
+
+def test_bad_filenames():
+    """
+    Test files that shouldn't get back a hit
+    TODO: Maybe learn hypothesis and use it here
+    """
+    crappy_filenames = [" ", ".", ".txt", "fasta.txt", "><", ":=).=:"]
+
+    for filename in crappy_filenames:
+        possible_importers = AutoDetectUtils.determine_possible_importers(
+            filename=filename
+        )
+        assert possible_importers is None
+
+
+def test_reasonable_filenames():
+    """
+    Test variants "reasonable" filenames that are in the mappings
+    Note: Some apps and mappings are not yet supported and are commented out
+    """
+
+    good_filenames = []
```

Comment:
It'd also be good to add in a file with numerous extensions (like the `.tar.gz` example) to prove that the auto detection doesn't get tripped up (and to ensure that if there are changes to auto detection in the future, they keep the current functionality).

Reply:
@ialarmedalien this is covered in the tests

Reply:
I can add this instead of the zip file test if you'd like

Reply:
Just checking that I am reading the code correctly -- you're going through each of the extension types (FASTA, FASTQ, MEDIA, FBA, etc.), and generating filenames with each of the extensions, e.g. `FASTA.fna`, `FASTA.fasta`, `FASTA.fa`, etc. In that case, the testing you already have is fine. 👍